core/domain/rights_manager.py

```diff
@@ -396,60 +581,155 @@ def _can_delete(self, rights_object):
+
+        Args:
+            activity_type: The type of activity. Either one of these:
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
+
+        Returns:
+            bool. Whether the user has a right to play an activity.
+        """
         activity_rights = _get_activity_rights(activity_type, activity_id)
         if activity_rights is None:
             return False
         return self._can_play(activity_rights)
 
     def can_view(self, activity_type, activity_id):
-        """Whether the user can view the editor page for this activity."""
+        """Returns whether this user has rights to view a given activity.
+
+        Args:
+            activity_type: The type of activity. Either one of these:
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
+
+        Returns:
+            bool. Whether the user has a right to view an activity.
+        """
         return self.can_play(activity_type, activity_id)
 
     def can_edit(self, activity_type, activity_id):
-        # TODO(sll): Add a check here for whether a user is banned or not,
-        # rather than having this check in the controller.
+        """Returns whether this user has rights to edit a given activity.
+
+        TODO(sll): Add a check here for whether a user is banned or not,
+        rather than having this check in the controller.
+
+        Args:
+            activity_type: The type of activity. Either one of these:
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
+
+        Returns:
+            bool. Whether the user has a right to edit an activity.
+        """
         activity_rights = _get_activity_rights(activity_type, activity_id)
         if activity_rights is None:
             return False
         return self._can_edit(activity_rights)
 
     def can_delete(self, activity_type, activity_id):
+        """Returns whether this user has rights to delete a given activity.
+
+        Args:
+            activity_type: The type of activity. Either one of these:
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
+
+        Returns:
+            bool. Whether the user has a right to delete an activity.
+        """
         activity_rights = _get_activity_rights(activity_type, activity_id)
         if activity_rights is None:
             return False
         return self._can_delete(activity_rights)
 
     def can_change_private_viewability(
             self, activity_type, activity_id):
-        """Note that this requires the activity in question to be private."""
+        """Returns whether this user has change private viewablility right.
```

Comment:
This sounds a little complicated. I am going with "Returns whether this user is allowed to change the viewability of a private activity.".